EddystoneBeacon.cpp

```diff
@@ -117,3 +144,35 @@ unsigned char EddystoneBeacon::compressURI(const char* uri, char *compressedUri,
 void EddystoneBeacon::loop() {
   this->poll();
 }
+/* loop() for TLM below:**/
+float temperature;
+int B=3975;                  //B value of the thermistor
+float resistance;
+unsigned long pdu_count = 0;
+void EddystoneBeacon::tempLoop(int analogPin) {
+  this->poll();
+  pdu_count++;
+  /** temperature **/
+  pinMode(analogPin,INPUT);
+  int a = analogRead(analogPin);
+  resistance=(float)(1023-a)*10000/a; //get the resistance of the sensor;
+  temperature=1/(log(resistance/10000)/B+1/298.15)-273.15;//convert to temperature via datasheet&nbsp;;
+  buffer_1[4] = ((float2fix(temperature))&0x0000FF00)>>8; //LSB
+  buffer_1[5] = (float2fix(temperature))&0x000000FF;  //MSB
+  //temperature
+  
+  unsigned long val = millis()/100;
+  ulong2adv(buffer,10, val);  //UP-TIME SINCE POWER ON OR REBOOT
+  ulong2adv(buffer,6, pdu_count);  //PDU COUNT
+
+  _bleCharacteristic.setValue(buffer, len);
+}
+void EddystoneBeacon::ulong2adv(uint8_t* adv, int off, unsigned long val) {
```

Comment:
@sandeepmistry 
I have removed `ulong2adv()` and then used memcpy as follows:
1.a new array called `newData[12]` will hold new values from the sensors.
2.using `memcpy()`,copied the `newData` to `buffer` from index 2 to 13.
3.Then setTLMData() looks as below.

void EddystoneBeacon::setTLMData(float temp, float bLevel) {
  pdu_count++;
  uint8_t newData[12];
  newData[0] = ((float2fix(bLevel))&0x0000FF00)>>8; //LSB
  newData[1] = (float2fix(bLevel))&0x000000FF;  //MSB
  newData[2] = ((float2fix(temp))&0x0000FF00)>>8; //LSB
  newData[3] = (float2fix(temp))&0x000000FF;  //MSB
  int off = 7;
  for (int i = 0; i < 4; ++i) {
    uint8_t bval = pdu_count & 0xff;
    newData[off] = bval;
    val >>= 8;
    off--;
  }
  val = millis()/100;
  off = 11;
  for (int i = 0; i < 4; ++i) {
    uint8_t bval = val & 0xff;
    newData[off] = bval;
    val >>= 8;
    off--;
  }
  memcpy(buffer + 2,newData, 12);
  _bleCharacteristic.setValue(buffer, len);
}